libpromises/verify_classes.c

```diff
@@ -147,20 +147,34 @@ static bool SelectClass(EvalContext *ctx, const Rlist *list, const Promise *pp)
         PromiseRef(LOG_LEVEL_ERR, pp);
         return false;
     }
-    assert(list);
+    else if (count == 1) /* There is no point of computation if there is only 
+                          * one item on a list. */
+    {
+        if (IsVarList(RlistScalarValue(list)))
+        {
+            Log(LOG_LEVEL_VERBOSE,
+                "select_class: Can not expand list '%s' for setting class.",
```

Comment:
This is handled exactly the same way as you have empty list to choose from (couple lines up in the same file). There won't be explicit message that evaluation will be skipped but we are not having this for any EvalExpression case so the behavior is consistent with what we are already having.